src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java

```diff
@@ -597,6 +599,38 @@ public void testAddRemoteUrl() throws Exception {
         assertEquals(upstreamRepoURL, gitClient.getRemoteUrl("upstream"));
     }
 
+    @Test
+    public void testAutocreateFailsOnMultipleMatchingOrigins() throws Exception {
+        File repoRootTemp = tempFolder.newFolder();
+        GitClient gitClientTemp = Git.with(TaskListener.NULL, new EnvVars()).in(repoRootTemp).using(gitImplName).getClient();
+        gitClientTemp.init();
```

Comment:
Seems like it is missing the initial commit that was added in the GitAPITestCase version of the test.  Maybe this would work (didn't compile it, just offering something based on javadoc exploration)?
```suggestion
        gitClientTemp.init();
        FilePath gitClientFilePath = gitClientTemp.getWorkTree()
        FilePath gitClientTempFile = gitClientFilePath.createTextTempFile("aPre", ".txt", "file contents");
        gitClientTemp.add(".");
        gitClientTemp.commit("Added " + gitClientTempFile.toURI().toString());
```

Reply:
Yes, you are right @MarkEWaite, I was wondering to add an empty commit which created an failure in CliGit but JGit and JGitApache were successful on that. This would be the right solution. Thank you.